From: Jan Blunck <jblunck@suse.de>
Subject: Provide empty DASD UID for FBA disks

This patch adds an empty DASD UID for FBA disks which don't support PAV.

Signed-off-by: Jan Blunck <jblunck@suse.de>
---
 drivers/s390/block/dasd_fba.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

Index: linux-2.6.16/drivers/s390/block/dasd_fba.c
===================================================================
--- linux-2.6.16.orig/drivers/s390/block/dasd_fba.c
+++ linux-2.6.16/drivers/s390/block/dasd_fba.c
@@ -125,6 +125,13 @@ static int
 dasd_fba_check_characteristics(struct dasd_device *device)
 {
 	struct dasd_fba_private *private;
+	struct dasd_uid uid = {
+		.alias = 0,
+		.vendor = "",
+		.serial = "",
+		.ssid = 0,
+		.unit_addr = 0,
+	};
 	struct ccw_device *cdev = device->cdev;	
 	void *rdc_data;
 	int rc;
@@ -159,6 +166,9 @@ dasd_fba_check_characteristics(struct da
 		    ((private->rdc_data.blk_bdsa *
 		      (private->rdc_data.blk_size >> 9)) >> 11),
 		    private->rdc_data.blk_size);
+
+	rc = dasd_set_uid(device->cdev, &uid);
+
 	return 0;
 }
 
